### DISCUSSION TITLE
How can I create global tag handles?

### DISCUSSION THREAD

**Discussion by hw762** [SanderMertens/flecs#896] [Q&A] [answered]
I tried to do `ECS_COMPONENT_DECLARE(SomeTag)` followed by `ECS_TAG(ecs, SomeTag)`. Unfortunately, this does not seem to work, as it fails an assertion: `flecs.c: 7856: assert: ecs_id_is_valid(world, id) INVALID_PARAMETER`. Is it possible to create global tag handles?

**Comment by copygirl** [comment, marked as answer]
If you want to forward-declare an entity that's not a component (created with `ECS_COMPONENT`) then you can use `ECS_DECLARE`.

As seen in the [fwd_declare_component example](https://github.com/SanderMertens/flecs/blob/9b7e7ed6985bfd9b255580a8a6b431f39ebaca90/examples/c/entities/fwd_declare_component/src/main.c#L19-L25):
```c
// To forward declare entities created with ECS_ENTITY, ECS_TAG or ECS_PREFAB,
// use ECS_DECLARE.
ECS_DECLARE(Wizard);
```
